api/v1alpha3/awscluster_types.go

```diff
@@ -82,6 +82,11 @@ type AWSClusterSpec struct {
 	// Bastion contains options to configure the bastion host.
 	// +optional
 	Bastion Bastion `json:"bastion"`
+
+	// SecretBackend is an optional string allowing the user to specify either Secrets Manger or SSM Parameter Storage.
+	// Only possible input is 'ssm', any other entry will default to Secrets Manager.
+	// +optional
```

Comment:
It might be worth adding extra kubebuilder validation here. Perhaps something like
```go
// +kubebuilder:default=secret-manager
// +kubebuilder:validation:Enum=secret-manager;parameter-store
```

Reply:
Nit: secrets-manager, not secret-manager
```
// +kubebuilder:default=secrets-manager
// +kubebuilder:validation:Enum=secrets-manager;parameter-store
```

Reply:
Additionally, I feel more confortable about moving this into the Cloudinit section of the AWSMachine spec. I understand that it's a bit more verbose, but it keeps a cleaner separation of things that relate to the overall cluster and those that relate to machines.

In addition, as we add support for Ignition, none of these methods will work, so we're strongly tying the AWSCluster resource to the Linux distribution here, which feels wrong.

Reply:
@randomvariable that's a good place to put it. I see that there is an option `EnableSecureSecretsManager` already present. Should this be replaced by a variable type that can encompass S3, Secrets Manager, and SSM?

Reply:
Ideally we would, but we can't replace it until the API is revved to v1alpha4, which will be 0.7.0.

Maybe change the description of `EnableSecureSecretsManager` to be to describe enabling the use of an external service in general, and then add something like `SecureSecretsBackend` to select on type.